pkgs/development/tools/literate-programming/Literate/default.nix

```diff
@@ -1,28 +1,48 @@
-{ lib, stdenv, fetchFromGitHub, ldc, dub }:
+{
+  lib,
+  buildDubPackage,
+  fetchFromGitHub,
+}:
 
-stdenv.mkDerivation {
+buildDubPackage {
   pname = "Literate";
   version = "unstable-2021-01-22";
 
   src = fetchFromGitHub {
     owner = "zyedidia";
     repo = "Literate";
     rev = "7004dffec0cff3068828514eca72172274fd3f7d";
-    sha256 = "sha256-erNFe0+FlrslEENyO/YxYQbmec0voK31UWr5qVt+nXQ=";
+    hash = "sha256-erNFe0+FlrslEENyO/YxYQbmec0voK31UWr5qVt+nXQ=";
     fetchSubmodules = true;
   };
 
-  buildInputs = [ ldc dub ];
+  # as there aren't any non-local dub dependencies, this file just has any empty list
+  dubDeps = ./deps.nix;
 
-  HOME = "home";
+  # generate the actual .d source files defined in .lit files
+  preBuild = ''
+    make d-files
+  '';
 
-  installPhase = "install -D bin/lit $out/bin/lit";
+  doCheck = true;
```

Comment:
Not sure how the other derivation wrappers do this but if this is this generally desirable for dub packages, perhaps it could be enabled by default.